### PR TITLE
Add Tensor OBB

### PR DESCRIPTION
**Pull request by yuecideng**
Summary:
1. Add `OrientedBoundingBox` in tensor geometry.
2. Change properties access method for pybind:
```python
# Original.
min_bound = aabb.get_min_bound()
rotation = obb.get_rotation()
# Currently, return read only property.
min_bound = aabb.min_bound
rotation = obb.rotation
```
3. Use box center as default arguments for `Scale` and `Rotate`.
4. Add orthogonality check for rotation setting of obb.
5. Fix some docs error related to aabb.

---
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d/5389)

### CODE REVIEW

**Comment by yuecideng** [comment]
Enable type casting to part of both aabb and obb functions (eg. `set_xxx`, `translate`, ...) and add warning message for notification. @yxlao @reyanshsolis  Is this modification looks good to you?

File: cpp/open3d/t/geometry/BoundingVolume.cpp [hunk 1]
```diff
@@ -132,34 +154,47 @@ void AxisAlignedBoundingBox::SetColor(const core::Tensor &color) {
...
+    color_ = color.To(GetDtype());
 }
 
 AxisAlignedBoundingBox &AxisAlignedBoundingBox::Translate(
         const core::Tensor &translation, bool relative) {
     core::AssertTensorDevice(translation, GetDevice());
     core::AssertTensorShape(translation, {3});
-    core::AssertTensorDtype(translation, GetDtype());
+    if (translation.GetDtype() != GetDtype()) {
+        PrintWarningMessageFromTensor("translation", translation.GetDtype(),
```
[the remark this answers is not in the document]
Comment (reyanshsolis) [reply cpp/open3d/t/geometry/BoundingVolume.cpp:165]: Skip warning, silently convert.

File: cpp/open3d/t/geometry/BoundingVolume.cpp [hunk 2]
```diff
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
...
+                     core::TensorKey::Index(1)}) *
+            extent_.GetItem({core::TensorKey::Index(1)}) * 0.5;
+    const core::Tensor z_axis =
+            rotation_.GetItem(
+                    {core::TensorKey::Slice(core::None, core::None, core::None),
+                     core::TensorKey::Index(2)}) *
+            extent_.GetItem({core::TensorKey::Index(2)}) * 0.5;
+
+    core::Tensor points = core::Tensor::Zeros({8, 3}, GetDtype(), GetDevice());
+    points.SetItem(core::TensorKey::Index(0),
```
Comment (reyanshsolis) [inline cpp/open3d/t/geometry/BoundingVolume.cpp:474]: To investigate a cleaner initialization possibility.

File: cpp/open3d/t/geometry/BoundingVolume.cpp [hunk 3]
```diff
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
...
+    points.SetItem(core::TensorKey::Index(7),
+                   center_ + x_axis + y_axis - z_axis);
+    return points;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Translate(
+        const core::Tensor &translation, bool relative) {
+    core::AssertTensorDevice(translation, GetDevice());
+    core::AssertTensorShape(translation, {3});
+    if (translation.GetDtype() != GetDtype()) {
```
Comment (reyanshsolis) [inline cpp/open3d/t/geometry/BoundingVolume.cpp:497]: Skip warning, silently convert.

File: cpp/open3d/t/geometry/BoundingVolume.cpp [hunk 4]
```diff
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
...
+    }
+    return *this;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Rotate(
+        const core::Tensor &rotation,
+        const utility::optional<core::Tensor> &center) {
+    core::AssertTensorDevice(rotation, GetDevice());
+    core::AssertTensorShape(rotation, {3, 3});
+    if (rotation.GetDtype() != GetDtype()) {
```
Comment (reyanshsolis) [inline cpp/open3d/t/geometry/BoundingVolume.cpp:516]: Skip warning, silently convert.

File: cpp/open3d/t/geometry/BoundingVolume.cpp [hunk 5]
```diff
@@ -289,6 +324,346 @@ AxisAlignedBoundingBox AxisAlignedBoundingBox::FromLegacy(
...
+    }
+
+    return *this;
+}
+
+OrientedBoundingBox &OrientedBoundingBox::Transform(
+        const core::Tensor &transformation) {
+    core::AssertTensorDevice(transformation, GetDevice());
+    core::AssertTensorShape(transformation, {4, 4});
+    if (transformation.GetDtype() != GetDtype()) {
```
Comment (reyanshsolis) [inline cpp/open3d/t/geometry/BoundingVolume.cpp:549]: Skip warning, silently convert.

**Review by reyanshsolis — CHANGES_REQUESTED** [review]

File: cpp/open3d/t/geometry/BoundingVolume.cpp (hunk 2 above)
Comment (yuecideng) [reply cpp/open3d/t/geometry/BoundingVolume.cpp:474]: How about the following way:

```cpp
const t::geometry::AxisAlignedBoundingBox aabb(GetExtent() * -0.5, GetExtent() * 0.5);
return aabb.GetBoxPoints().Matmul(GetRotation()).Add(GetCenter());
 ```

**Comment by yuecideng** [comment]
Remove the warning message when casting input tensor.

**Comment by yxlao** [comment]
I have the same issue building the `benchmark` executable as the CI reports when CUDA is on. The `master` branch builds fine. I am looking into it.

**Comment by yxlao** [comment]
> I have the same issue building the `benchmark` executable as the CI reports when CUDA is on. The `master` branch builds fine. I am looking into it.

Ok, it's due to missing `;`.

**Review by yxlao — APPROVED** [review]

**Comment by yxlao** [comment]
LGTM after CI.

**Review by reyanshsolis — APPROVED** [review]

**Pull request merged**